src/core/datastore/utils/utils.js

```diff
@@ -0,0 +1,38 @@
+import { Metadata } from '../../metadata';
+import { Client } from '../../client';
+
+export function buildCollectionUrl(collectionName, id, restAction) {
+  let result = `appdata/${Client.sharedInstance().appKey}/${collectionName}`;
+  if (id) {
+    result += `/${id}`;
+  }
+  if (restAction) {
+    result += `/${restAction}`;
+  }
+  return result;
+}
+
+export function generateEntityId(length = 24) {
+  const chars = 'abcdef0123456789';
+  let objectId = '';
+
+  for (let i = 0, j = chars.length; i < length; i += 1) {
+    const pos = Math.floor(Math.random() * j);
+    objectId += chars.substring(pos, pos + 1);
+  }
+
+  return objectId;
+}
+
+export function isNotEmpty(object) {
+  return !!object && (object.length > 0 || Object.keys(object).length > 0);
+}
+
+export function isEmpty(object) {
```

Comment:
Why not use https://lodash.com/docs/#isEmpty?

Reply:
Hah, yeah... `isNotEmpty()` (above) used to be a different check, but I guess it has grown into lodash's `!isEmpty()`.

👍 

Btw, didn't you say we were moving away from lodash?

Reply:
I have been looking into ways we can decrease the bundles size of our SDKs. Lodash continues to add a lot of code that we may not need (even with tree shaking). It is quite possible we could write the necessary utility functions as part of the SDK and remove the dependency on Lodash.

The downside to this is Lodash is well tested and we know we can trust it where as rewriting these utilities would require a lot more testing to cover edge cases. 

For now we still have it as a dependency so lets continue to use it until we decide not too.

Reply:
Done in a later PR.